src/MonoMod.RuntimeDetour/Hook.cs

```diff
@@ -454,8 +454,55 @@ public Hook(MethodBase source, Delegate target, IDetourFactory factory, DetourCo
         MethodInfo IDetour.InvokeTarget => realTarget;
 
 
-        private readonly MethodInfo trampoline;
-        MethodBase IDetour.NextTrampoline => trampoline;
+        private sealed class TrampolineData : IDetourTrampoline, IDisposable {
+
+            private readonly object LOCK = new object();
```

Comment:
I'm a little surprised that the editorconfig didn't cause a warning for this. Generally, I'd name it like a normal local (i.e. `syncRoot` or similar) or just lock on `this`, given that we know exactly who will lock on this type.

Reply:
Got it, will change that.